### PR TITLE
Add `os.eventfd`, `os.eventfd_read` and `os.eventfd_write` to Linux for Python 3.10+

### PR DESCRIPTION
**Pull request by aminalaee**
Ref: https://docs.python.org/3/library/os.html#os.eventfd

### CODE REVIEW

**Review by AlexWaygood — DISMISSED** [review]
LGTM, thanks! Happy to merge once you've taken this out of draft :)

**Comment by aminalaee** [comment]
@AlexWaygood I think we can add more of the functions related to this in the same PR, if that's fine by you.

**Comment by AlexWaygood** [comment]
Sure!

File: stdlib/os/__init__.pyi [hunk 1]
```diff
@@ -1072,3 +1080,4 @@ if sys.version_info >= (3, 12) and sys.platform == "linux":
     CLONE_SYSVSEM: int
     CLONE_THREAD: int
     CLONE_VM: int
+    def unshare(flags: int) -> None: ...
```
Comment (aminalaee) [inline stdlib/os/__init__.pyi:1083]: This was related to https://github.com/python/typeshed/pull/10757 but I thought doesn't required a separate PR.

File: stdlib/os/__init__.pyi [hunk 2]
```diff
@@ -1057,6 +1058,14 @@ if sys.version_info >= (3, 12) and sys.platform == "win32":
     def listmounts(volume: str) -> list[str]: ...
     def listvolumes() -> list[str]: ...
 
+if sys.version_info >= (3, 10) and sys.platform == "linux":
+    EFD_CLOEXEC: int
+    EFD_NONBLOCK: int
+    EFD_SEMAPHORE: int
+    def eventfd(initval: int, flags: int = 524288) -> FileDescriptor: ...
+    def eventfd_read(fd: FileDescriptor) -> int: ...
+    def eventfd_write(fd: FileDescriptor, value: int) -> int: ...
```
[the remark this answers is not in the document]
Comment (aminalaee) [reply stdlib/os/__init__.pyi:1067]: You are right.

**Review by AlexWaygood — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**